### PR TITLE
docker-test: handle race condition error

### PR DESCRIPTION
**Pull request by rst0git**
There is a race condition in docker/containerd that causes docker to occasionally fail when starting a container from a checkpoint immediately after the checkpoint has been created. This problem is unrelated to criu and has been reported in https://github.com/moby/moby/issues/42900

### CODE REVIEW

File: scripts/ci/docker-test.sh [hunk 1]
```diff
@@ -75,17 +75,37 @@ checkpoint_container () {
...
 		docker logs cr || true
 		cat $CRIU_LOG || true
 		dmesg
 		docker ps
 		exit 1
-	}
+}
+
+declare -i max_restore_container_tries=3
+current_iteration=
```
Comment (avagin) [inline scripts/ci/docker-test.sh:88]: why is it declared outside of restore_container?

File: scripts/ci/docker-test.sh [hunk 2]
```diff
@@ -75,17 +75,37 @@ checkpoint_container () {
...
+
+restore_container () {
+	CHECKPOINT_NAME=$1
+
+	docker start --checkpoint "$CHECKPOINT_NAME" cr 2>&1 | tee log || {
+		# FIXME: There is a race condition in docker/containerd that causes
+		# docker to occasionally fail when starting a container from a
+		# checkpoint immediately after the checkpoint has been created.
+		# https://github.com/moby/moby/issues/42900
+		if [ "$current_iteration" -gt "$max_restore_container_tries" ]; then
```
Comment (avagin) [inline scripts/ci/docker-test.sh:98]: https://github.com/checkpoint-restore/criu/actions/runs/6948976416/job/18929387032?pr=2299
```
+ '[' '' -gt 3 ']'
1800
./docker-test.sh: line 100: [: : integer expression expected
```

Comment (rst0git) [reply scripts/ci/docker-test.sh:98]: Thanks for pointing this out. I opened a pull request to fix this: https://github.com/checkpoint-restore/criu/pull/2302

**Comment by rst0git** [comment]
> There is a race condition in docker/containerd that causes docker to occasionally fail when starting a container from a checkpoint immediately after the checkpoint has been created.

It looks like this problem has now been fixed upstream (https://github.com/moby/moby/pull/47456).

**Pull request merged**